Vagrantfile

```diff
@@ -48,13 +48,40 @@ Vagrant.configure('2') do |config|
 
   # Networking configuration.
   config.vm.hostname = vconfig['vagrant_hostname']
-  config.vm.network :private_network,
-    ip: vconfig['vagrant_ip'],
-    auto_network: vconfig['vagrant_ip'] == '0.0.0.0' && Vagrant.has_plugin?('vagrant-auto_network')
 
+  if vconfig['vagrant_ip'] == 'dhcp'
+    # Automatically assign IP
+    config.vm.network :private_network, type: 'dhcp'
+
+    # Use assigned IP with vagrant-hostmanager
+    if Vagrant.has_plugin?('vagrant-hostmanager')
```

Comment:
This section feels so complicated though... basically going from a few lines to 22 lines with some regex and shell commands thrown in (if I'm reading it correctly)?

Reply:
Is the only issue if you use `vagrant-hostmanager`? E.g. does it work currently if you have `vagrant-hostsupdater` instead?

Reply:
The scary looking `config.hostmanager.ip_resolver` proc is what makes `vagrant-hostmanager` pick up the DHCP assigned IPs from within the VM and then adds them to `/etc/hosts`.

Reply:
No, vagrant-hostsupdater doesn't work unless they add support for the same `ip_resolver` functionality. In which case this logic would probably be the same anyways. https://github.com/cogitatio/vagrant-hostsupdater/issues/56

Reply:
@geerlingguy if you prefer not having this logic how would you feel about leaving this section as:

```rb
if vconfig['vagrant_ip'] == 'dhcp'
  # Automatically assign IP. See docs for integration with `vagrant-hostmanager`
  config.vm.network :private_network, type: 'dhcp'
elsif ...
```

And moving the snippet there and users can include it in their `Vagrantfile.local`.

Reply:
Since September we've been using this at work on 10+ VM's without `vagrant-vbguest`. No issues so far.